### PR TITLE
[dynamo] Added support for math.radians on ints with dynamic shapes

### LINKED ISSUE #114218: math.radians with integer input creates a graph break for dynamic shapes
### 🐛 Describe the bug

```python
import itertools
import math

import torch


input = 90
expected = math.pi / 2


for type, dynamic in itertools.product([float, int], [False, True]):
    torch._dynamo.reset()
    cfn = torch.compile(fullgraph=True, dynamic=dynamic)(math.radians)

    try:
        torch.testing.assert_close(cfn(type(input)), expected)
        result = "PASS"
    except:
        result = "FAIL"
    print(f"{result} {type=}, {dynamic=}")
```

```
PASS type=<class 'float'>, dynamic=False
PASS type=<class 'float'>, dynamic=True
PASS type=<class 'int'>, dynamic=False
FAIL type=<class 'int'>, dynamic=True
```

<details><summary>Traceback</summary>
<p>

```
Traceback (most recent call last):
  File "/home/philip/git/pytorch/vision/main_dynamo.py", line 15, in <module>
    torch.testing.assert_close(cfn(type(input)), expected)
  File "/home/philip/.conda/envs/pytorch-vision-dev/lib/python3.8/site-packages/torch/_dynamo/eval_frame.py", line 482, in _fn
    return fn(*args, **kwargs)
  File "/home/philip/.conda/envs/pytorch-vision-dev/lib/python3.8/site-packages/torch/_dynamo/eval_frame.py", line 646, in catch_errors
    return callback(frame, cache_entry, hooks, frame_state)
  File "/home/philip/.conda/envs/pytorch-vision-dev/lib/python3.8/site-packages/torch/_dynamo/convert_frame.py", line 383, in _convert_frame_assert
    compiled_product = _compile(
  File "/home/philip/.conda/envs/pytorch-vision-dev/lib/python3.8/site-packages/torch/_dynamo/convert_frame.py", line 645, in _compile
    guarded_code = compile_inner(code, one_graph, hooks, transform)
  File "/home/philip/.conda/envs/pytorch-vision-dev/lib/python3.8/site-packages/torch/_dynamo/utils.py", line 244, in time_wrapper
    r = func(*args, **kwargs)
  File "/home/philip/.conda/envs/pytorch-vision-dev/lib/python3.8/site-packages/torch/_dynamo/convert_frame.py", line 562, in compile_inner
    out_code = transform_code_object(code, transform)
  File "/home/philip/.conda/envs/pytorch-vision-dev/lib/python3.8/site-packages/torch/_dynamo/bytecode_transformation.py", line 1033, in transform_code_object
    transformations(instructions, code_options)
  File "/home/philip/.conda/envs/pytorch-vision-dev/lib/python3.8/site-packages/torch/_dynamo/convert_frame.py", line 151, in _fn
    return fn(*args, **kwargs)
  File "/home/philip/.conda/envs/pytorch-vision-dev/lib/python3.8/site-packages/torch/_dynamo/convert_frame.py", line 527, in transform
    tracer.run()
  File "/home/philip/.conda/envs/pytorch-vision-dev/lib/python3.8/site-packages/torch/_dynamo/symbolic_convert.py", line 2123, in run
    super().run()
  File "/home/philip/.conda/envs/pytorch-vision-dev/lib/python3.8/site-packages/torch/_dynamo/symbolic_convert.py", line 818, in run
    and self.step()
  File "/home/philip/.conda/envs/pytorch-vision-dev/lib/python3.8/site-packages/torch/_dynamo/symbolic_convert.py", line 781, in step
    getattr(self, inst.opname)(inst)
  File "/home/philip/.conda/envs/pytorch-vision-dev/lib/python3.8/site-packages/torch/_dynamo/symbolic_convert.py", line 470, in wrapper
    return inner_fn(self, inst)
  File "/home/philip/.conda/envs/pytorch-vision-dev/lib/python3.8/site-packages/torch/_dynamo/symbolic_convert.py", line 1252, in CALL_FUNCTION_EX
    self.call_function(fn, argsvars.items, kwargsvars.items)
  File "/home/philip/.conda/envs/pytorch-vision-dev/lib/python3.8/site-packages/torch/_dynamo/symbolic_convert.py", line 652, in call_function
    self.push(fn.call_function(self, args, kwargs))
  File "/home/philip/.conda/envs/pytorch-vision-dev/lib/python3.8/site-packages/torch/_dynamo/variables/lazy.py", line 90, in realize_and_forward
    return getattr(self.realize(), name)(*args, **kwargs)
  File "/home/philip/.conda/envs/pytorch-vision-dev/lib/python3.8/site-packages/torch/_dynamo/variables/torch.py", line 599, in call_function
    tensor_variable = wrap_fx_proxy(
  File "/home/philip/.conda/envs/pytorch-vision-dev/lib/python3.8/site-packages/torch/_dynamo/variables/builder.py", line 1280, in wrap_fx_proxy
    return wrap_fx_proxy_cls(target_cls=TensorVariable, **kwargs)
  File "/home/philip/.conda/envs/pytorch-vision-dev/lib/python3.8/site-packages/torch/_dynamo/variables/builder.py", line 1539, in wrap_fx_proxy_cls
    unimplemented(
  File "/home/philip/.conda/envs/pytorch-vision-dev/lib/python3.8/site-packages/torch/_dynamo/exc.py", line 193, in unimplemented
    raise Unsupported(msg)
torch._dynamo.exc.Unsupported: torch.* op returned non-Tensor float call_function <built-in function radians>

from user code:
   File "/home/philip/.conda/envs/pytorch-vision-dev/lib/python3.8/site-packages/torch/_dynamo/external_utils.py", line 17, in inner
    return fn(*args, **kwargs)

Set TORCH_LOGS="+dynamo" and TORCHDYNAMO_VERBOSE=1 for more information


You can suppress this exception and fall back to eager by setting:
    import torch._dynamo
    torch._dynamo.config.suppress_errors = True
```

</p>
</details> 

### Versions

2.2.0.dev20231120+cpu

cc @ezyang @msaroufim @wconstab @bdhirsh @anijain2305 @zou3519

### PR DESCRIPTION
**Pull request by vfdev-5**
Fixes #114218

```
import math
import torch


def func(x, a):
    b = math.floor(a + 0.5)
    b = math.radians(a) + b
    y = x + b
    return y


cfunc = torch.compile(func, dynamic=True, fullgraph=True, backend="eager")
x = torch.tensor([0, 1, 2, 3], dtype=torch.float32)
a = 12

out = cfunc(x, a)
```

```
[2023-11-29 18:10:08,385] [0/0] torch._dynamo.output_graph.__graph_code: [DEBUG] TRACED GRAPH
[2023-11-29 18:10:08,385] [0/0] torch._dynamo.output_graph.__graph_code: [DEBUG]  ===== __compiled_fn_0 =====
[2023-11-29 18:10:08,385] [0/0] torch._dynamo.output_graph.__graph_code: [DEBUG]  <eval_with_key>.0 class GraphModule(torch.nn.Module):
[2023-11-29 18:10:08,385] [0/0] torch._dynamo.output_graph.__graph_code: [DEBUG]     def forward(self, L_a_ : torch.SymInt, s1 : torch.SymInt, L_x_ : torch.Tensor):
[2023-11-29 18:10:08,385] [0/0] torch._dynamo.output_graph.__graph_code: [DEBUG]         l_a_ = L_a_
[2023-11-29 18:10:08,385] [0/0] torch._dynamo.output_graph.__graph_code: [DEBUG]         l_x_ = L_x_
[2023-11-29 18:10:08,385] [0/0] torch._dynamo.output_graph.__graph_code: [DEBUG]
[2023-11-29 18:10:08,385] [0/0] torch._dynamo.output_graph.__graph_code: [DEBUG]         # File: check_math_ops.py:7, code: b = math.floor(a + 0.5)
[2023-11-29 18:10:08,385] [0/0] torch._dynamo.output_graph.__graph_code: [DEBUG]         add = l_a_ + 0.5
[2023-11-29 18:10:08,385] [0/0] torch._dynamo.output_graph.__graph_code: [DEBUG]         floor = math_floor(add);  add = None
[2023-11-29 18:10:08,385] [0/0] torch._dynamo.output_graph.__graph_code: [DEBUG]
[2023-11-29 18:10:08,385] [0/0] torch._dynamo.output_graph.__graph_code: [DEBUG]         # File: /pytorch/torch/_dynamo/polyfill.py:28, code: return math.pi / 180.0 * x
[2023-11-29 18:10:08,385] [0/0] torch._dynamo.output_graph.__graph_code: [DEBUG]         mul = 0.017453292519943295 * l_a_;  l_a_ = None
[2023-11-29 18:10:08,385] [0/0] torch._dynamo.output_graph.__graph_code: [DEBUG]
[2023-11-29 18:10:08,385] [0/0] torch._dynamo.output_graph.__graph_code: [DEBUG]         # File: check_math_ops.py:9, code: b = math.radians(a) + b
[2023-11-29 18:10:08,385] [0/0] torch._dynamo.output_graph.__graph_code: [DEBUG]         add_1 = mul + floor;  mul = floor = None
[2023-11-29 18:10:08,385] [0/0] torch._dynamo.output_graph.__graph_code: [DEBUG]
[2023-11-29 18:10:08,385] [0/0] torch._dynamo.output_graph.__graph_code: [DEBUG]         # File: check_math_ops.py:13, code: y = x + b
[2023-11-29 18:10:08,385] [0/0] torch._dynamo.output_graph.__graph_code: [DEBUG]         y = l_x_ + add_1;  l_x_ = add_1 = None
[2023-11-29 18:10:08,385] [0/0] torch._dynamo.output_graph.__graph_code: [DEBUG]         return (y,)
[2023-11-29 18:10:08,385] [0/0] torch._dynamo.output_graph.__graph_code: [DEBUG]
[2023-11-29 18:10:08,385] [0/0] torch._dynamo.output_graph.__graph_code: [DEBUG]
```

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

**Review by lamroger — COMMENTED** [review]
Some minor suggestions to help make it clearer. 

I'm new to this so take with a grain of salt...

File: torch/_dynamo/variables/torch.py [hunk 1]
```diff
@@ -290,12 +290,26 @@ def call_function(
...
+            or unspec_python_args
+            or any(isinstance(x, SymNodeVariable) for x in args)
+        ):
             # constant fold
             return ConstantVariable.create(
                 self.as_python_constant()(
-                    *[x.as_python_constant() for x in args],
-                    **{k: v.as_python_constant() for k, v in kwargs.items()},
+                    *[
+                        x.as_python_constant()
```
[the remark this answers is not in the document]
Comment (vfdev-5) [reply torch/_dynamo/variables/torch.py]: @lamroger thanks for suggestions, I reworked the PR to propagate sym variable within math.radians.

**Review by lezcano — APPROVED** [review]

**Comment by lezcano** [comment]
@lamroger  We have to implement these functions in a way that's traceable with symints/symfloats, and the simplest way to do so is via polyfills. Do you have a better idea for how to do this?

**Pull request closed without merging**